tests/cli/fixtures/cli_old_projects.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2021 Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku CLI fixtures for old project management."""
+from pathlib import Path
+
+import pytest
+from git import Repo
+
+
+def clone_compressed_repository(base_path, name):
+    """Decompress and clone a repository."""
+    import tarfile
+
+    compressed_repo_path = Path(__file__).parent / ".." / ".." / "data" / f"{name}.tar.gz"
+    working_dir = base_path / name
+
+    bare_base_path = working_dir / "bare"
+
+    with tarfile.open(compressed_repo_path, "r") as fixture:
+        fixture.extractall(str(bare_base_path))
+
+    bare_path = bare_base_path / name
+    repository_path = working_dir / "repository"
+    repository = Repo(bare_path, search_parent_directories=True).clone(repository_path)
+
+    return repository
+
+
+@pytest.fixture(
+    params=["old-datasets-v0.3.0.git", "old-datasets-v0.5.0.git", "old-datasets-v0.5.1.git", "test-renku-v0.3.0.git"]
+)
+@pytest.mark.usefixtures("request", "tmp_path")
+def old_project(request, tmp_path):
```

Comment:
As I understand this, `pytest.mark.usefixtures` only makes sense when a fixture needs to be run for a test to work, but the test itself doesn't access the fixture (going by https://docs.pytest.org/en/stable/fixture.html#use-fixtures-in-classes-and-modules-with-usefixtures ). E.g. if the fixture changes cwd to a temp directory, but the test doesn't care about what the path is.

So adding `usefixtures` and then also specifying the fixture as an argument doesn't do anything if I understand this correctly. Is there something else that needs this set?

Reply:
When it comes to tests, you're absolutely right (but here we have chained fixtures). This is more of a refactoring trick and could be removed if we don't want to refactor fixtures further. One thing which I see it could be useful to keep it in case if we want to move to multiple `conftest.py` with different configurations, otherwise I can remove it.

Reason why I added it is because during refactoring, pytest collector sometimes did not know about chained fixtures (cause I would move things around), and in those cases I've used this as a hinting mechanism to the collector that this is suppose to be a fixture, even though at the time it was not included and recognized in the runtime as one, this allowed me to easier move things around.

Reply:
Done.